compiler/rustc_resolve/src/late.rs

```diff
@@ -1158,8 +1174,20 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
                 return;
             }
 
-            if let LifetimeRibKind::Item = rib.kind {
-                break;
+            match rib.kind {
+                LifetimeRibKind::Item => break,
+                LifetimeRibKind::ConstGeneric => {
+                    self.emit_non_static_lt_in_const_generic_error(lifetime);
+                    self.r.lifetimes_res_map.insert(lifetime.id, LifetimeRes::Error);
+                    return;
+                }
+                LifetimeRibKind::AnonConst
+                    if self.maybe_emit_forbidden_non_static_lifetime_error(lifetime) =>
```

Comment:
`#[feature]` attributes aren't supposed to change behavior, only to emit or not emit stability errors.
The previous logic looked more in line with that strategy, it emitted the error and then continued as usual.